### PR TITLE
Deactivate guard

### PR DESCRIPTION
**Pull request by the-greenman**
This is an implementation of a popup message to warn a user that some changes to the workflow have not been saved.

### CODE REVIEW

**Comment by the-greenman** [comment]
https://user-images.githubusercontent.com/166254/151370624-aa931998-a853-4de3-9b2a-8b75a7778045.mp4

**Comment by the-greenman** [comment]
Now with a working and tested guard to prevent loss of changes

File: angular.json [hunk 1]
```diff
@@ -108,7 +108,7 @@
                 },
                 "fonts": false
               },
-              "sourceMap": false,
+              "sourceMap": true,
```
Comment (the-greenman) [inline angular.json:111]: This setting disabled the ability to debug! 
Can move into a separate branch if necessary

File: src/app/services/dirty-dialog.service.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@angular/core';
+import { Observable, of } from 'rxjs';
+
+/**
+ * Async modal dialog service
+ * DialogService makes this app easier to test by faking this service.
```
Comment (lukestanley) [inline src/app/services/dirty-dialog.service.ts:6]: Do you mean "DirtyDialogService"?
@the-greenman

File: src/app/services/dirty-dialog.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+describe('DirtyDialogService', () => {
+  let service: DirtyDialogService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({});
+    service = TestBed.inject(DirtyDialogService);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
```
Comment (lukestanley) [inline src/app/services/dirty-dialog.service.spec.ts:14]: Is this useful? @the-greenman

Comment (the-greenman) [reply src/app/services/dirty-dialog.service.spec.ts:14]: I'm wondering about these auto generated stubs. They do provide an essential bit of test coverage. They do actually make sure that the object gets created. 

In this case I don't actually know how to test the current functionality of this service outside of Cypress, but functionality may evolve. 

Getting back to the Angular default standard of always having a spec test for a file may be a good objective.

File: src/app/services/dirty-dialog.service.ts (hunk 2 above)
Comment (the-greenman) [reply src/app/services/dirty-dialog.service.ts:6]: Was an old comment anyway. Removed

**Review by lukestanley — APPROVED** [review]

**Pull request merged**